openedx/core/djangoapps/credit/signals.py

```diff
@@ -70,3 +81,133 @@ def listen_for_grade_calculation(sender, username, grade_summary, course_key, de
                     api.set_credit_requirement_status(
                         username, course_id, 'grade', 'grade', status="failed", reason={}
                     )
+
+
+def tag_course_content_with_partition_scheme(course_key, partition_scheme):
+    """ Create user partitions with provided partition scheme and tag credit
+    blocks for the given course with those user partitions.
+
+    Args:
+        course_key (CourseKey): Identifier for the course.
+        partition_scheme (str): Name of the user partition scheme
+
+    """
+    # TODO: Dynamically update user_id (actual course publisher)
+    user_id = 1
+
+    # get user partition with provided partition scheme
+    user_partition = UserPartition.get_scheme(partition_scheme)
+    if user_partition is None:
+        # log and raise exception 'NoSuchUserPartitionError' if no matching
+        # user partition exists
+        log.error(
+            u'No user partition found with scheme "%s". Exiting tagging course content with user partition.',
+            partition_scheme
+        )
+        raise NoSuchUserPartitionError
+
+    access_control_credit_xblocks = _get_credit_xblocks_for_access_control(course_key)
+    all_user_partitions = []
+    for block in access_control_credit_xblocks:
+        # for now we are only entertaining verification partition scheme
+        if partition_scheme == VERIFICATION_SCHEME:
+            # TODO: Break code in small reusable code
+            # create group configuration for VerificationPartitionScheme
+            group_configuration = _verification_partition_group_configuration(user_partition, block)
+            all_user_partitions.append(group_configuration)
+
+            # update 'group_access' field of gating xblock with groups
+            # 'verified_allow' and 'verified_deny' of newly created group
+            # configuration
+            access_dict = {
+                group_configuration.id: [
+                    group_configuration.get_group('verified_allow').id,
```

Comment:
Instead of hardcoded string we can use the VerificationPartitionScheme.VERIFIED_ALLOW.